django_prices_openexchangerates/__init__.py

```diff
@@ -72,22 +45,38 @@ def convert_price(price, to_currency, get_rate=get_conversion_rate):
         conversion_rate = 1 / rate
     else:
         conversion_rate = rate
-    conversion = CurrencyConversion(
-        base_currency=price.currency,
-        to_currency=to_currency,
-        rate=conversion_rate)
-    return conversion.apply(price)
+    return conversion_rate
 
 
-def exchange_currency(price, to_currency, get_rate=get_conversion_rate):
+def exchange_currency(
+        base: T, to_currency: str, get_rate=get_rate_from_db) -> T:
     """
-    Exchanges Price or PriceRange to the specified currency
+    Exchanges Money, TaxedMoney and their ranges to the specified currency.
+    get_rate parameter is a callable taking single argument (target currency)
+    that returns proper conversion rate
     """
-    if isinstance(price, PriceRange):
-        return PriceRange(
-            exchange_currency(price.min_price, to_currency),
-            exchange_currency(price.max_price, to_currency))
-    if price.currency != BASE_CURRENCY:
-        # Convert to default currency
-        price = convert_price(price, BASE_CURRENCY, get_rate=get_rate)
-    return convert_price(price, to_currency, get_rate=get_rate)
+    if base.currency == to_currency:
+        return base
+    if base.currency != BASE_CURRENCY and to_currency != BASE_CURRENCY:
+        # Exchange to base currency first
+        base = exchange_currency(base, BASE_CURRENCY, get_rate=get_rate)
+
+    conversion_rate = get_conversion_rate(base.currency, to_currency, get_rate)
+
+    if isinstance(base, Money):
+        return Money(base.amount * conversion_rate, currency=to_currency)
+    if isinstance(base, MoneyRange):
+        return MoneyRange(exchange_currency(base.start, to_currency,
+                                            get_rate=get_rate),
```

Comment:
Please don't do line continuation like that. Fold after each opening parenthesis.